include/revng/AutoEnforcer/AutoEnforcerContext.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <initializer_list>
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringMap.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/Casting.h"
+
+#include "revng/AutoEnforcer/KindsRegistry.h"
+#include "revng/AutoEnforcer/SavableObject.h"
+
+namespace AutoEnforcer {
+class NamedGlobal {
+public:
+  NamedGlobal(llvm::StringRef Name, SavableObjectBase &Obj) :
+    Name(Name), Obj(&Obj) {}
+
+  llvm::StringRef getName() const { return Name; }
+
+  SavableObjectBase &getObj() const { return *Obj; }
+
+private:
+  llvm::StringRef Name;
+  SavableObjectBase *Obj;
+};
+
+class Context {
+public:
+  explicit Context(KindsRegisty registry) : KRegistry(std::move(registry)) {}
```

Comment:
KindRegistry seems to me a support data structure that could be populated internally given the root of the kind tree.
Why put the burden on the `Context` creator to do that?

Reply:
because this allows in general to configure auto enforcer with different kinds,and kinds cannot be polluted by other libraries that end up being linked in. In particulars allows to selects particular kinds for particular tets

Reply:
Suppose you have:
```graphviz
digraph {
  A -> B -> C;
  B -> D;
}
```
Are you saying that there is an interest in leaving out part of a subtree? For instance, leaving out `D`?
Sounds a bit odd.
For testing purposes you can simply have a dedicated root that is unaffected by any library.

Just to be clear, what I'm suggesting is to have the constructor take a `Kind *` and internally construct a `KindsRegistry` of the subtree that can be reached from that one.

Reply:
the issue with that the root of this tree that would be talen by the autoenforcer is "weird" when used in enforcer, its children would not all have the same granularity